### PR TITLE
Add and Delete note

### PR DESCRIPTION
**Pull request by Jay-Gadhiya**
Test Link: https://cheer-notes-p2y6wh73x-jay-gadhiya.vercel.app/

Feature : Add note, Delete note

Context: notesActions-context.jsx
Reducer :  notesActions-reducer.jsx

### CODE REVIEW

File: src/Utilities-Functions/addNote.jsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import axios from "axios";
+
+
+const addNote = async (note, authState, noteDispatch, setUserNote) => {
+    try {
+        const res = await axios.post("/api/notes", { note }, { headers : { authorization: authState.token } }); 
+        setUserNote(pre => ({ ...pre, title : "", content : "" }));
+        noteDispatch({type : "ADD_NOTE", payload : res.data.notes });
+        console.log(res);
```
Comment (RahullRawat) [inline src/Utilities-Functions/addNote.jsx]: You can remove these unwanted `console.log`

File: src/Utilities-Functions/addNote.jsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        
+    } catch (error) {
+        console.log(err);
```
Comment (RahullRawat) [inline src/Utilities-Functions/addNote.jsx]: You can remove these unwanted `console.log`

File: src/Utilities-Functions/deleteNote.jsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import axios from "axios";
+
+const deleteNote = async (item, authState, noteDispatch) => {
+    try {
+        const res = await axios.delete(`/api/notes/${item._id}`, { headers : { authorization: authState.token } }); 
+        noteDispatch({type : "DELETE_NOTE", payload : res.data.notes });
+        
+    } catch (error) {
+        console.log(error);
```
Comment (RahullRawat) [inline src/Utilities-Functions/deleteNote.jsx]: You can remove these unwanted `console.log`

**Review by RahullRawat — APPROVED** [review]
**Just one suggestion that user shouldn't be able to add an empty note.** 

- Other than this everything looks good to me
- Good naming conventions and well formatted code

Great work @Jay-Gadhiya

File: src/Contexts/notesActions-context.jsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { notesReducer } from "../Reducers/notesActions-reducer";
+
+const { createContext, useContext, useReducer } = require("react");
+
```
Comment (gopalbbharadva) [inline src/Contexts/notesActions-context.jsx:4]: Suggestion: without require we can also import it by using normal "import"

**Review by jeetbhalani15 — APPROVED** [review]
Remove logs after debuging.
Use Context & reducer are working fine.
Clean Name & readable code.

File: src/Pages/Home/Home.jsx [hunk 5]
```diff
@@ -1,18 +1,38 @@
...
+        try {
+            const res = await axios.post("/api/notes", { note }, { headers : { authorization: authState.token } }); 
+            setUserNote(pre => ({ ...pre, title : "", content : "" }));
+            noteDispatch({type : "ADD_NOTE", payload : res.data.notes });
+            console.log(res);
+            
+        } catch (error) {
+            console.log(err);
+        }
+    }
```
Comment (gopalbbharadva) [inline src/Pages/Home/Home.jsx]: Improvement: Add note only if response status is 201. remove unused consoles.

File: src/Reducers/notesActions-reducer.jsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
+
+const notesReducer = (state, action) => {
+    
+    switch (action.type) {
+        case "ADD_NOTE":
+            return {...state, notes : [...action.payload] };
+        
+        case "DELETE_NOTE":
+            return {...state, notes : [...action.payload] };
```
Comment (AtulPant2704) [inline src/Reducers/notesActions-reducer.jsx]: Over here you can just write :- `return {...state, notes : action.payload };` and that would also work, why to destructure unnecessarily.

**Review by gopalbbharadva — CHANGES_REQUESTED** [review]
Bug: Note is adding one or both of the inputs are not given. just add validation required on both inputs.

**Review by AtulPant2704 — APPROVED** [review]
Other than the one change mentioned, rest all looks good. One point that Rahul has already mentioned do work on it, in upcoming PRs atleast `Title `should be provided by the user. Good naming convention with properly formatted data, made it really easy to review.

**Comment by Jay-Gadhiya** [comment]
> **Just one suggestion that user shouldn't be able to add an empty note.**
> 
> * Other than this everything looks good to me
> * Good naming conventions and well formatted code
> 
> Great work @Jay-Gadhiya

I Implemented it, thanks for the suggestion.

**Pull request merged**